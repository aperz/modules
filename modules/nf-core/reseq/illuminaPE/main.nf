// TODO nf-core: If in doubt look at other nf-core/modules to see how we are doing things! :)
//               https://github.com/nf-core/modules/tree/master/modules/nf-core/
//               You can also ask for help via your pull request or on the #modules channel on the nf-core Slack workspace:
//               https://nf-co.re/join
// TODO nf-core: A module file SHOULD only define input and output files as command-line parameters.
//               All other parameters MUST be provided using the "task.ext" directive, see here:
//               https://www.nextflow.io/docs/latest/process.html#ext
//               where "task.ext" is a string.
//               Any parameters that need to be evaluated in the context of a particular sample
//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
//               list (`[]`) instead of a file can be used to work around this issue.

process RESEQ {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/reseq:1.1--py38hc35fec1_3':
        'biocontainers/reseq:1.1--py38hc35fec1_3' }"


    input:
    path(fasta)
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.fastq.gz")         , emit: fastqgz
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    reseq illuminaPE \\
        ${args} \\
        -j ${task.cpus} \\
        -r ${fasta} \\
        -b ${bam} \\
        -1 ${prefix}.1.fastq.gz \\
        -2 ${prefix}.2.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        reseq: \$(reseq --version |& sed 's/ReSeq version //g')
    END_VERSIONS
    """
}
