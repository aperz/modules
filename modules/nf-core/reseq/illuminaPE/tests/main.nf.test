nextflow_process {

    name "Test Process RESEQ"
    script "../main.nf"
    process "RESEQ"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "reseq"

    test("homo sapiens - fasta + bam") {
        when {
            process {
                """
                input[0] = [
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test_bam', single_end:false ],
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }
}
