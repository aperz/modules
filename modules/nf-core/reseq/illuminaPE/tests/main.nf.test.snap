{
    "homo sapiens - fasta + bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        [
                            "test_bam.1.fastq.gz:md5,9ed3da4bef24ea8c552133500dd7692f",
                            "test_bam.2.fastq.gz:md5,7e1c980ad299916e3cd1ac5fd7aabb4e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,9d1dfeb98e4dbaa40904ea2e57459acc"
                ],
                "fastqgz": [
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        [
                            "test_bam.1.fastq.gz:md5,9ed3da4bef24ea8c552133500dd7692f",
                            "test_bam.2.fastq.gz:md5,7e1c980ad299916e3cd1ac5fd7aabb4e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9d1dfeb98e4dbaa40904ea2e57459acc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T20:05:54.529717944"
    }
}